### PR TITLE
[TC] Improve the coverage of the Scrollview widget

### PR DESCRIPTION
**Pull request by HunseopJeong**
[Issue] N/A
[Problem] Scrollview coverage is low
[Solution] Add new test cases

Signed-off-by: Hunseop Jeong <hs85.jeong@samsung.com>

### CODE REVIEW

**Comment by HunseopJeong** [comment]
Before:
image

After:
image

File: tests/js/profile/wearable/widget/wearable/Scrollview/Scrollview.js [hunk 1]
```diff
@@ -44,6 +44,85 @@
 			helpers.restoreStub(ns.support.shape, "circle");
 		});
 
+		QUnit.test("_init", 4, function (assert) {
+			var element = document.getElementById("scrollview-page-w"),
+				scrollview = new Scrollview();
+
+			helpers.stub(Scrollview.prototype, "_setBouncingEffect", function () {
```
Comment (TomaszLukawskiSam) [inline tests/js/profile/wearable/widget/wearable/Scrollview/Scrollview.js]: This stub works but we rather make stab from instance method instead prototype method.

File: tests/js/profile/wearable/widget/wearable/Scrollview/Scrollview.js [hunk 2]
```diff
@@ -44,6 +44,85 @@
...
+
+		QUnit.test("showBouncingEffect & hideBouncingEffect", 5, function (assert) {
+			var element = document.getElementById("scrollview-page-w"),
+				scrollview = new Scrollview();
+
+			helpers.stub(Scrollview.prototype, "_setBouncingEffect", function () {
+				assert.ok(true, "_setBouncingEffect should be called");
+			});
+
+			scrollview.element = element;
```
Comment (TomaszLukawskiSam) [inline tests/js/profile/wearable/widget/wearable/Scrollview/Scrollview.js]: these 3 lines seems not needed.

You need stub for scrollview.scroller: this.scroller.getBoundingClientRect().height

File: tests/js/profile/wearable/widget/wearable/Scrollview/Scrollview.js (hunk 1 above)
Comment (HunseopJeong) [reply tests/js/profile/wearable/widget/wearable/Scrollview/Scrollview.js]: Thank you for review.
Done.

File: tests/js/profile/wearable/widget/wearable/Scrollview/Scrollview.js (hunk 2 above)
Comment (HunseopJeong) [reply tests/js/profile/wearable/widget/wearable/Scrollview/Scrollview.js]: If I remove the 3 lines, I got following message.
```
TypeError: Can not read property 'drag' of undefined
```
If there is no process for initializing the widget, fail to get the bouncingEffect object.

Comment (TomaszLukawskiSam) [reply tests/js/profile/wearable/widget/wearable/Scrollview/Scrollview.js]: Hi :),
Implementation of `showBouncingEffect` method is simple.
```
			prototype.showBouncingEffect = function (place) {
				var bouncingEffect = this.bouncingEffect;

				if (place === "end") {
					bouncingEffect.drag(0, -this.scroller.getBoundingClientRect().height);
				} else {
					bouncingEffect.drag(0, 0);
				}
			};
```
This is unit test. Its purpose to test only this method.

You need two mocks.
one:
```
scrollview.bouncingEffect = {
   drag: function (arg1, arg2) {
       assert.ok(true, "Method bouncingEffect has been called");
       assert.equal(arg1, 0, "Method bouncingEffect: first argument equals 0");
   }
}
```
two:
```
scrollview.scroller = {
    getBoundingClientRect: function () {
         assert.ok(true, "Method getBoundingClientRect has been called");
         return {
              height: 100; // a sample value;
         }
    }
}
```

Comment (HunseopJeong) [reply tests/js/profile/wearable/widget/wearable/Scrollview/Scrollview.js]: I see. I added your suggestion.

**Review by TomaszLukawskiSam — APPROVED** [review]
Ok.

**Review by lslachciak — APPROVED** [review]
good,
nit for a future tests:

I would check also if 
assert.equal(scrollview.dragging, false, "dragging is false"
before scrollview._start()

**Comment by HunseopJeong** [comment]
> good,
> nit for a future tests:
> 
> I would check also if
> assert.equal(scrollview.dragging, false, "dragging is false"
> before scrollview._start()

We cannot check the ' assert.equal(scrollview.dragging, false, "dragging is false") '
Because scrollview.dragging is not initialized before calling the start().

**Review by DohyungLim — COMMENTED** [review]
LGTM.

**Review by DohyungLim — APPROVED** [review]

**Review by singa2000 — APPROVED** [review]
It's good

**Pull request merged**